psh/cp/cp.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Phoenix-RTOS
+ *
+ * cp - copy file
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "../psh.h"
+
+#define SIZE_BUFF 256
+
+
+static void psh_cpinfo(void)
+{
+	printf("copy file");
+}
+
+
+static void psh_cp_help(const char *prog)
+{
+	printf("Usage: %s [options] SOURCE TARGET\n", prog);
+	printf("  -h:  shows this help message\n");
+}
+
+
+static int psh_cp(int argc, char **argv)
+{
+	FILE *src = NULL, *dst = NULL;
+	int fdsrc, fddst;
+	unsigned char buff[SIZE_BUFF];
+	char *filename, *destpath;
+	size_t destlen, filelen, cnt;
+	struct stat stat;
+	int c, retval = EXIT_SUCCESS, eexit = 0;
+
+	while ((c = getopt(argc, argv, "h")) != -1) {
+		switch (c) {
+			case 'h':
+				psh_cp_help(argv[0]);
+				eexit = 1;
+				break;
+			default:
+				psh_cp_help(argv[0]);
+				retval = EXIT_FAILURE;
+				eexit = 1;
+				break;
+		}
+	}
+
+	if (argc - optind != 2) {
+		psh_cp_help(argv[0]);
+		retval = EXIT_FAILURE;
+		eexit = 1;
+	}
+
+	while (eexit == 0) {
+		fdsrc = open(argv[optind], O_RDONLY);
+		if (fdsrc < 0) {
+			retval = EXIT_FAILURE;
+			fprintf(stderr, "cp: could not open source file\n");
+			break;
+		}
+
+		if (fstat(fdsrc, &stat) < 0) {
+			close(fdsrc);
+			retval = EXIT_FAILURE;
+			fprintf(stderr, "cp: stat failed\n");
+		}
+
+		if (S_ISDIR(stat.st_mode) != 0) {
```

Comment:
No need to `!= 0`, S_ISDIR returns bool, see:
```c
#define S_ISDIR(m)  (((m) & S_IFMT) == S_IFDIR)  /* Directory */
```

Reply:
Thanks!